src/test/java/bio/terra/service/snapshot/SnapshotTest.java

```diff
@@ -285,4 +295,68 @@ public void snapshotByFullViewHappyPathTest() throws Exception {
         assertEquals("all 5 relationships come through", snapshot.getRelationships().size(), 5);
     }
+
+    @Test
+    public void snapshotACLTest() throws Exception {
+        DatasetModel dataset = dataRepoFixtures.getDataset(steward(), datasetId);
+
+        String datasetName = dataset.getName();
+
+        logger.info("---- Dataset ACLs before snapshot create-----");
+        int datasetAclCount = fetchSourceDatasetAcls(datasetName);
+
+        //-------------------Create Snapshot----------------
+        SnapshotRequestModel requestModel =
+            jsonLoader.loadObject("ingest-test-snapshot-fullviews.json", SnapshotRequestModel.class);
+        // swap in the correct dataset name (with the id at the end)
+        requestModel.getContents().get(0).setDatasetName(datasetName);
+        SnapshotSummaryModel snapshotSummary =
+            dataRepoFixtures.createSnapshotWithRequest(steward(),
+                datasetName,
+                profileId,
+                requestModel);
+        TimeUnit.SECONDS.sleep(10);
+        createdSnapshotIds.add(snapshotSummary.getId());
+        SnapshotModel snapshot = dataRepoFixtures.getSnapshot(steward(), snapshotSummary.getId());
+        assertEquals("new snapshot has been created", snapshot.getName(), requestModel.getName());
+        assertEquals("all 5 relationships come through", snapshot.getRelationships().size(), 5);
+
+        // fetch ACLs
+        logger.info("---- Dataset ACLs after snapshot create-----");
+        int dataset_plus_snapshot_count = fetchSourceDatasetAcls(datasetName);
+        assertThat("There should be more ACLs on the dataset after snapshot create",
+            dataset_plus_snapshot_count, greaterThan(datasetAclCount));
+
+        //-----------delete snapshot------------
+        dataRepoFixtures.deleteSnapshot(steward(), snapshotSummary.getId());
+
+        // Give ACLs a second to update in big query
+        TimeUnit.SECONDS.sleep(10);
+
+        logger.info("---- Dataset ACLs after snapshot delete-----");
+        int dataset_minus_snapshot_acl_count = fetchSourceDatasetAcls(datasetName);
+        assertEquals("We should be back to the same number of ACLs on the dataset after snapshot delete",
+            datasetAclCount, dataset_minus_snapshot_acl_count);
+    }
+
+    private int fetchSourceDatasetAcls(String datasetName) {
+        String projectId = System.getenv("GOOGLE_CLOUD_DATA_PROJECT");
+        bigQuery = BigQueryOptions.newBuilder()
+            .setProjectId(projectId)
+            .build()
+            .getService();
+
+        // Fetch BQ Dataset
+        String bqDatasetName = bigQueryPdao.prefixName(datasetName);
+        Dataset bq_dataset = bigQuery.getDataset(bqDatasetName);
+
+        // fetch ACLs
+        List<Acl> acls = bq_dataset.getAcl();
+        int aclCount = acls.size();
+
+        acls.forEach(acl -> {
+            logger.info("Acl: {}", acl);
+        });
```

Comment:
You don't need the `{}` s if the function body is a single statement
```suggestion
        acls.forEach(acl -> logger.info("Acl: {}", acl));
```